app/src/main/java/org/mifos/mobile/ui/transaction/TransactionActivity.kt

```diff
@@ -0,0 +1,87 @@
+package org.mifos.mobile.ui.transaction
+
+import android.os.Bundle
+import android.text.TextUtils
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import kotlinx.android.synthetic.main.activity_transaction.*
+import org.json.JSONObject
+import org.mifos.mobile.R
+import org.mifos.mobile.models.entity.*
+import org.mifos.mobile.ui.activities.base.BaseActivity
+import org.mifos.mobile.utils.Toaster
+import javax.inject.Inject
+
+class TransactionActivity : BaseActivity(), TransactionContract.View, View.OnClickListener {
+
+    @Inject
+    internal lateinit var transactionPresenter: TransactionPresenter
+    var jsonob: JSONObject? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_transaction)
+        showBackButton()
+        setToolbarTitle("Send Money")
+        activityComponent?.inject(this)
+        transactionPresenter.attachView(this)
+        send.setOnClickListener(this)
+        if(intent.getStringExtra("JSON")!=null) {
+            jsonob = JSONObject(intent.getStringExtra("JSON"))
+            MSISDNtext.setText(jsonob?.getString("mobile"))
+        }
+    }
+
+    override fun onClick(view: View) {
+
+        when (view.id) {
+            R.id.send -> transaction()
+        }
+
+    }
+
+    fun transaction() {
```

Comment:
can we use a better method name? applies here and other places where relevant.
Give meaningful method names such that even a novice guy can understand the underlying functionality of it without reading the code.